tm_cmd/tm_manifest.py

```diff
@@ -19,39 +19,66 @@
 
 def _cleanup_sysarg(sysargv, parsearg):
     """
-        Remove "argparse" defined arguments from sys.argv list.
+        Remove "argparse" defined arguments from sys.argv list as well as any
+    other ars that are not in cmdlookup.
     Mixing those two together causing confusion in the "cmdlookup"
     function calls.
     :param 'sysargv': [list] of arguments in "sys.argv"
     :param 'parsearg': [dict] of arguments set by "argparse" module.
     """
-    for arg in parsearg.keys():
-        overlap = '--%s' % (arg)
-        if overlap in sysargv:
-            sysargv.remove(overlap)
+    to_remove = [] #arguments to be removed from sysargv
+    for arg in sysargv:
+        # remove all dashes and = signes from the argument
+        arg_name = arg.strip('-').split('=')[0]
+        if arg not in cmdlookup:
+            # removing directly from sysargv will cause problems. Thus, save
+            # what needs to be removed and do it after this loop is done.
+            to_remove.append(arg)
+            continue
+
+        if arg_name in parsearg:
+            sysargv.remove(arg)
+            to_remove.append(arg)
+
+    # at this point, it is safe to remove from sysargv list, since it is outside
+    # the sysargv loop.
+    for arg in to_remove:
+        if arg in sysargv:
+            sysargv.remove(arg)
 
 
 def main(args):
-    if len(sys.argv) < 2:
+    _cleanup_sysarg(sys.argv, args)
```

Comment:
Why are you preprocessing sys.argv before argparser gets it?   Won't argparser do  most of the work you want?   Maybe you just need to move the test for missing directives to after argparse...

Reply:
This is due to the way "tmcmd" module is structured. I'll write a README for this at some point.

Reply:
OK but if you leave HPE you have to support this module forever.